src/preamble.js

```diff
@@ -2221,6 +2221,19 @@ function integrateWasmJS(Module) {
       Module['asm'] = exports;
       Module["usingWasm"] = true;
     }
+
+    // User shell pages can write their own Module.instantiateWasm = function(imports, successCallback) callback
+    // to manually instantiate the Wasm module themselves. This allows pages to run the instantiation parallel
+    // to any other async startup actions they are performing.
+    if (Module['instantiateWasm']) {
+      try {
+        return Module['instantiateWasm'](info, receiveInstance);
```

Comment:
Ok, changed the run dependency to happen automatically: https://github.com/kripken/emscripten/pull/5055/commits/639da1260649220b64f3f240fcd06ce7332ea4f8